python/cudf/cudf/_lib/pylibcudf/column.pxd

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.column.column_view cimport column_view
+from cudf._lib.cpp.types cimport offset_type, size_type
+
+from .gpumemoryview cimport gpumemoryview
+from .types cimport DataType
+
+
+cdef class Column:
+    cdef:
+        # Core data
+        DataType data_type
+        size_type size
+        gpumemoryview data
+        gpumemoryview mask
+        size_type null_count
+        offset_type offset
+        # TODO: Is there a more efficient container I could use than a list
+        # here? For now defaulting to List[gpumemoryview]
```

Comment:
```suggestion
        # here? For now defaulting to List[Column]
```

Reply:
A list is probably fine. We can't use a tuple as I'm pretty sure there are places in cuDF where we have to mutate the `children` attribute of a column.

Reply:
Agreed. I'll stick to this for now.